lightning/src/util/chacha20.rs

```diff
@@ -249,6 +251,29 @@ mod real_chacha {
 				self.offset += count;
 			}
 		}
+
+		pub fn process_inline(&mut self, input_output: &mut [u8]) {
```

Comment:
Are you relying on LLVM judgment to inline this function ? Also this function process in-place, not inline, that's a bit confusing.. (or what the term exactly to mean you don't access heap for processing operation, can't remember)

Reply:
This is less about bytecode optimization and more about not having to allocate an output buffer if we don't need the input any longer. In place is a much, much better name to describe that, so renamed this.